titanic.py

```diff
@@ -93,12 +93,46 @@ def maximum_energy_delta(bqm):
                      for j in bqm.iter_neighbors(i))
                for i in bqm.iter_variables())
 
-def run_demo():
+def mutual_information_bqm(dataset, features):
+    """Build a QUBO that maximizes MI between survival and a subset of features"""
+    variables = ((feature, -mutual_information(prob(dataset[['survived', feature]].values), 1))
+                 for feature in features)
+    interactions = ((f0, f1, -conditional_mutual_information(prob(dataset[['survived', f0, f1]].values), 1, 2))
+                    for f0, f1 in itertools.permutations(features, 2))
+    return dimod.BinaryQuadraticModel(variables, interactions, 0, dimod.BINARY)
+
+def add_combination_penalty(bqm, k, penalty):
+    """Create a new BQM with an addition k-combination penalty"""
+    kbqm = dimod.generators.combinations(bqm.variables, k, strength=penalty)
+    kbqm.update(bqm)
+    return kbqm
+
+def mutual_information_feature_selection(dataset, features):
+    """Run the MIFS algoeith on D-Wave"""
```

Comment:
Spelling

Reply:
Also "on D-Wave" should be something like "on a D-Wave system." or "on a QPU solver."

Reply:
"on a D-Wave solver" --> I appreciate that you tried to combine both proposals :-). But "a D-Wave system" and "a QPU solver" refer specifically to our QCs, and I think your docstring wanted to point that out (as opposed to running on HSS or a classical solver).
It's not crucial here but worth being aware of the terminology:
* HSS: "Leap's hybrid solvers" or more formally/first mention "Leap's quantum-classical hybrid solvers"  
* 2000Q and Advantage processors: prior to HSS we called these "D-Wave systems", but that term is less unambiguous now, so we're more often saying "QPU solvers"